tests/test_service.py

```diff
@@ -279,6 +279,54 @@ def test_tag_build_if_multiple_rules_are_matched(
                 ]
             })
 
+    @patch('message_tagging_service.tagging_service.retrieve_modulemd_content')
+    @patch('koji.ClientSession')
+    @patch('koji.read_config')
+    def test_tag_build_with_multiple_tags(
+            self, read_config, ClientSession, retrieve_modulemd_content):
+        read_config.return_value = koji_config_krb_auth
+
+        # Because rule file specifies that destination tag uses the value of
+        # requires.platform, and there are two of those values in moduelmd,
+        # the module build should be tagged with two tags.
+        retrieve_modulemd_content.return_value = dedent('''\
+            ---
+            document: modulemd
+            version: 2
+            data:
+              name: javapackages-tools
+              stream: 1
+              version: 1
+              context: c1
+              dependencies:
+              - buildrequires:
+                  platform: [f29]
+                requires:
+                  platform: [f29, f28]
```

Comment:
There is still a problem. For a stream expanded module, it should not have multiple streams in `requires` as well. For example the libgit2 module, it has dependencies in modulemd:

```yaml
dependencies:
  - buildrequires:
      platform: []
    requires:
      platform: []
```

It is expanded into 3 module builds for f28, f29 and f30, e.g.

```yaml
- buildrequires:
    platform: [f29]
  requires:
    platform: [f29]
```

Reply:
I'm not sure what you mean. Where would those 3 platforms come from? Is there a unit test we can add up verify this behavior?

Reply:
@tkdchen in this case, since the buildrequires and requires list aren't exactly the same, it would end up with one build that buildrequires f29 but would require either f29 or f28 at install time.

See the example in the spec:
https://github.com/fedora-modularity/libmodulemd/blob/master/spec.v2.yaml#L109

Reply:
Got it. Thanks Matt.